webapp/app.py

```diff
@@ -427,6 +427,46 @@ def markup(text):
     return markdown.markdown(text)
 
 
+@app.template_filter()
+def filtered_html_tags(content):
+    content = content.replace("<p>&nbsp;</p>", "")
+    allowed_tags = [
+        "iframe",
+        "h2",
+        "h3",
+        "h4",
+        "h5",
+        "h6",
+        "p",
+        "a",
+        "strong",
+        "ul",
+        "ol",
+        "li",
+        "i",
+        "em",
+        "br",
+    ]
+    allowed_attributes = {"iframe": allow_src}
+
+    return bleach.clean(
+        content,
+        tags=allowed_tags,
+        attributes=allowed_attributes,
+        strip=True,
+    )
+
+
+def allow_src(tag, name, value):
```

Comment:
- `tag` is not used anywhere? 
- Does it work this way? As far as I know, Python doesn't have hoisting, the function should be defined before?

Reply:
Bleach passes those three arguments implicitly when functions are called this way, we just don't use the `tag` argument in the function being called. And yes, it works.